### ISSUE TITLE
Microsoft.XmlSerializer.Generator too long commandline argument

### ISSUE THREAD

**Issue by danstur** [dotnet/corefx#34670]
I'm using the Microsoft.XmlSerializer.Generator nuget in a netstandard2.0 project which references about 20 NuPkgs. In the following manner:
```
  <ItemGroup>
    <PackageReference Include="Microsoft.XmlSerializer.Generator" Version="2.0.0" />
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.XmlSerializer.Generator" Version="2.0.0" />
  </ItemGroup>
```
So nothing out of the ordinary. During the build I get the following error:
```
[8>System.ComponentModel.Win32Exception (206): The filename or extension is too long
8>   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
8>   at System.Diagnostics.Process.Start()
8>   at Microsoft.DotNet.Cli.Utils.Command.Execute()
8>   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
8>   at Microsoft.DotNet.Cli.Program.Main(String[] args)
8>C:\Users\das.000\AppData\Local\NuGet\.nuget\microsoft.xmlserializer.generator\2.0.0\build\Microsoft.XmlSerializer.Generator.targets(15,5): warning MSB3073: The command "dotnet Microsoft.XmlSerializer.Generator "obj\Debug\netstandard20\StoreSR.Core.dll" --force --quiet --reference "C:\Users\das.000\AppData\Local\NuGet\.nuget\netstandard.library\2.0.3\build\netstandard2.0\ref\netstandard.dll;C:\Users\das.000\AppData\Lo.. and so on"](url)
```

The problem stems from passing the fully-qualified path of all reference assemblies (246 in my case) on the commandline which causes the commandline to become longer than the 32k limit. Looking at the code it doesn't seem like there's a workaround for this problem (apart from moving the nuget cache folder to a shorter path). 

Using a temporary file or stdin for passing references to the Generator executable would be one possible solution to this problem. Would such a pull request be accepted for the code? 

(Haven't provided the code to reproduce this, since the problem seems pretty obvious from the code itself - if you think you'd need it, I can certainly do it).

Dotnet SDK: 2.2.101
OS: Windows 10 1809 x64

**Comment by yujayee** [comment]
Hello @danstur, could you share with me exactly how long is your dotnet Microsoft.XmlSerializer.Generator command?

**Comment by danstur** [comment]
@yujayee Yes there more than 32,768 characters in the built command line. The error message seems to be misleading, the filename itself was far from any limit (even the 260 or so of MAX_PATH). 

Not a problem if you use a separate contracts assembly without many references which is probably more common, but it'd be nice if we weren't forced to do this.

**Comment by huanwu** [comment]
@danstur, by our design, we will pass all the references of your project to be serialized to the XmlSerializer.Generator so it can resolve the type defined in those references. It might cause this issue if you have too many references and the path is too long. Is it possible for you to put the type to be serialized into a separate project that can use less references or use the shorter path?

**Comment by yujayee** [comment]
This is fixed by pr #36095
